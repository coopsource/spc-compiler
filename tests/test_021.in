77 59 144 116
160 91
155
62
279
36
