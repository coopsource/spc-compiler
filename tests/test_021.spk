define function sort taking arguments a and n as
	set i to 1
	while i < n do
		set j to i + 1
		while j < n+1 do
			if element i of a > element j of a then
				set x to element i of a
				set element i of a to element j of a
				set element j of a to x
			else
				nothing
			end if
			set j to j + 1
		end while
		set i to i + 1
	end while

	print a
end function

define function main taking no arguments as
	set i to 1
	set element 1 of X to 0
	while i < 11 do
		read element i of X
		set i to i + 1
	end while

	call sort with X and 10
end function
