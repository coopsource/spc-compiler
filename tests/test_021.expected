[36, 59, 62, 77, 91, 116, 144, 155, 160, 279]
